### PR TITLE
feat: use fxart endpoint for drop

### LINKED ISSUE #9104: Implement new drop worker
### Is your feature request related to a problem?

Many issues comes from the fact that we use really hacky version of drops worker. 
It can be safely deprecated.

### Task

Your task is to implement new worker called [fxart](https://fxart.kodadot.workers.dev)

and it works bit differently than the waifu worker

[The whole README is here](https://github.com/kodadot/private-workers/tree/main/fxart)

But TL;DR user first need to allocate the "ticket" which assigns him/her a serial number that can be used

If it is paid (price > 0) drop  you need to wait for transaction **to be successfully finished** (this does not work now :))

Once you get blocknumber and txhash of the TX it is safe to call finish api.

For free drop you will allocate, (there will be step to sign something but we do not have design for that yet) and call finish.

### Describe alternatives you have considered

Other apis were untouched.

If something is unclear/missing please hit me up or open issue in private-worker repo

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #9104

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### CODE REVIEW

**Comment by preschian** [comment]
thanks and sorry Vik, for flooding messages on Telegram 🙇‍♂️

todos:
- [ ] I need 1 more free campaign to test this component, I guess? `<CollectionDropGenerative v-else-if="dropType === 'generative'" :drop="drop" />`
- [ ] need 1 endpoint to check claimed. ref: https://github.com/kodadot/private-workers/issues/45

cc @vikiival

**Comment by preschian** [comment]
https://github.com/kodadot/nft-gallery/blob/b9e4ae2fcd397bbc8ed96aa24b9b27a19e1fca49/pages/%5Bprefix%5D/drops/%5Bid%5D.vue#L3-L14

1. Currently, our drops have some types. Do we still use all of them? Related: https://github.com/kodadot/private-workers/pull/50
2. There is also `Boolean(drop.holds)` in that code section. Do we still need that? Related: https://github.com/kodadot/private-workers/issues/49

cc @vikiival @Jarsen136 

Currently, I only tested this component `<CollectionDropPaidGenerative />`

File: services/waifu.ts
Comment (vikiival) [inline services/waifu.ts:1]: please create new file :)

Comment (preschian) [reply services/waifu.ts:1]: just rename or need to create a new file?

Comment (yangwao) [reply services/waifu.ts:1]: I think `services/fxart.ts`

Comment (vikiival) [reply services/waifu.ts:1]: > just rename or need to create a new file?

new, I think some other parts of code are dependent on that :)

Comment (preschian) [reply services/waifu.ts:1]: like this? [76d8cc7](https://github.com/kodadot/nft-gallery/pull/9226/commits/76d8cc7464fb3df4bbff1b386b4bfa25ea5458d8)

**Comment by yangwao** [comment]
@preschian probably needs update as there was new drop page merged
- https://github.com/kodadot/nft-gallery/pull/9264

**Comment by vikiival** [comment]
This weeks drop is ready on both services

File: pages/[prefix]/drops/[id].vue [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
   <UnlockableCollectionBanner :drop="drop" />
-  <CollectionDropContainer v-if="dropType === 'paid'" :drop="drop" />
+  <!-- <CollectionDropContainer v-if="dropType === 'paid'" :drop="drop" /> -->
   <CollectionDropHolderOfGenerative
-    v-else-if="dropType === 'generative' && Boolean(drop.holds)"
+    v-if="dropType === 'holder' && Boolean(drop.holder_of)"
     :drop="drop" />
   <CollectionDropPaidGenerative
-    v-else-if="dropType === 'generative' && Number(drop.price)"
+    v-else-if="dropType === 'paid' && Number(drop.price)"
```
Comment (Jarsen136) [inline pages/[prefix]/drops/[id].vue:8]: We do need to check `holder_of` or `price`, because we have already independent type for different drops.

**Comment by Jarsen136** [comment]
> 2. There is also `Boolean(drop.holds)` in that code section. Do we still need that? Related: [No `holds` on fxart? private-workers#49](https://github.com/kodadot/private-workers/issues/49)

No, we dont need to check them on frontend side anymore.

File: pages/[prefix]/drops/[id].vue (hunk 1 above)
Comment (preschian) [reply pages/[prefix]/drops/[id].vue:8]: Did you mean `"we don't need to check"`?

Comment (Jarsen136) [reply pages/[prefix]/drops/[id].vue]: Yes, let remove them `&& Number(drop.price)` ` && Boolean(drop.holder_of)`

**Review by Jarsen136 — APPROVED** [review]

File: services/fxart.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  email: string
+  hash: string
+}
+export const getDropMintedStatus = async (alias: string, accountId: string) => {
+  return await api<DropMintedStatus>(`/drops/${alias}/accounts/${accountId}`, {
+    method: 'GET',
+  })
+}
+
+export const allocateCollection = async (body, id) => {
```
[the remark this answers is not in the document]
Comment (vikiival) [reply services/fxart.ts:51]:
```suggestion
export const allocateDrop = async (body, id) => {
```
?

**Review by vikiival — APPROVED** [review]
Minor, otherwise lgtm

**Comment by yangwao** [comment]
Thanks!
pay 150 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $150 @ 6.7 USD/DOT ~ 22.388 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0xdc66e4c4ce5f970a29ece900a8cba79eae2d0603f7ba8eb55117b1dc8771a430](https://polkadot.subscan.io/extrinsic/0xdc66e4c4ce5f970a29ece900a8cba79eae2d0603f7ba8eb55117b1dc8771a430)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by vikiival** [comment]
So we can use the worker for tomorrows drop

**Pull request merged**